lib/buffer.js

```diff
@@ -1049,9 +1040,8 @@ Buffer.prototype.writeFloatLE = function writeFloatLE(val, offset, noAssert) {
 Buffer.prototype.writeFloatBE = function writeFloatBE(val, offset, noAssert) {
   val = +val;
   offset = offset >>> 0;
-  if (!noAssert)
-    checkFloat(this, val, offset, 4);
-  binding.writeFloatBE(this, val, offset);
+  noAssert = !!noAssert;
+  binding.writeFloatBE(this, val, offset, noAssert);
```

Comment:
sorry, minor nit: do the boolean coercion in the call itself. e.g. `writeFloatBE(this, val, offset, !!noAssert);`